node/hyperbahn/handler.js

```diff
@@ -0,0 +1,223 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+/* jshint maxparams:5 */
+
+var assert = require('assert');
+var util = require('util');
+
+var TChannelJSON = require('../as/json');
+var TChannelEndpointHandler = require('../endpoint-handler');
+
+module.exports = HyperbahnHandler;
+
+function HyperbahnHandler(options) {
+    if (!(this instanceof HyperbahnHandler)) {
+        return new HyperbahnHandler(options);
+    }
```

Comment:
This pattern is useful under two conditions:

 - The constructor is not called often. (per server, not per request)
 - The constructor is part of the public interface.

Under those two conditions it's useful to hide the fact that `new` is required. This makes it easier for developers to use the interfaces exposed by your package without having to remember whether its a constructor or not.